src/hb-ot-metrics.cc

```diff
@@ -58,53 +57,48 @@ hb_ot_metrics_get (hb_font_t       *font,
-  if (font->face->table.TABLE->has_data ()) \
-  { \
-    if (position) *position = font->em_scale_##DIR (font->face->table.TABLE->ATTR); \
-    return true; \
-  } \
-  else return false
+  (font->face->table.TABLE->has_data () && \
+    ((position ? *position = font->em_scale_##DIR (font->face->table.TABLE->ATTR) : 0), true))
 
-  case HB_OT_METRICS_HORIZONTAL_ASCENDER:         GET_METRICS (OS2, sTypoAscender, y);
-  case HB_OT_METRICS_HORIZONTAL_DESCENDER:        GET_METRICS (OS2, sTypoDescender, y);
-  case HB_OT_METRICS_HORIZONTAL_LINE_GAP:         GET_METRICS (OS2, sTypoLineGap, y);
-  case HB_OT_METRICS_HORIZONTAL_CLIPPING_ASCENT:  GET_METRICS (OS2, usWinAscent, y);
-  case HB_OT_METRICS_HORIZONTAL_CLIPPING_DESCENT: GET_METRICS (OS2, usWinDescent, y);
-  case HB_OT_METRICS_VERTICAL_ASCENDER:           GET_METRICS (vhea, ascender, x);
-  case HB_OT_METRICS_VERTICAL_DESCENDER:          GET_METRICS (vhea, descender, x);
-  case HB_OT_METRICS_VERTICAL_LINE_GAP:           GET_METRICS (vhea, lineGap, x);
-  case HB_OT_METRICS_HORIZONTAL_CARET_RISE:       GET_METRICS (hhea, caretSlopeRise, y);
-  case HB_OT_METRICS_HORIZONTAL_CARET_RUN:        GET_METRICS (hhea, caretSlopeRun, y);
-  case HB_OT_METRICS_HORIZONTAL_CARET_OFFSET:     GET_METRICS (hhea, caretOffset, y);
-  case HB_OT_METRICS_VERTICAL_CARET_RISE:         GET_METRICS (vhea, caretSlopeRise, x);
-  case HB_OT_METRICS_VERTICAL_CARET_RUN:          GET_METRICS (vhea, caretSlopeRun, x);
-  case HB_OT_METRICS_VERTICAL_CARET_OFFSET:       GET_METRICS (hhea, caretOffset, x);
-  case HB_OT_METRICS_X_HEIGHT:                    GET_METRICS (OS2->get_v2 (), sxHeight, y);
-  case HB_OT_METRICS_CAP_HEIGHT:                  GET_METRICS (OS2->get_v2 (), sCapHeight, y);
-  case HB_OT_METRICS_SUBSCRIPT_EM_X_SIZE:         GET_METRICS (OS2, ySubscriptXSize, x);
-  case HB_OT_METRICS_SUBSCRIPT_EM_Y_SIZE:         GET_METRICS (OS2, ySubscriptYSize, y);
-  case HB_OT_METRICS_SUBSCRIPT_EM_X_OFFSET:       GET_METRICS (OS2, ySubscriptXOffset, x);
-  case HB_OT_METRICS_SUBSCRIPT_EM_Y_OFFSET:       GET_METRICS (OS2, ySubscriptYOffset, x);
-  case HB_OT_METRICS_SUPERSCRIPT_EM_X_SIZE:       GET_METRICS (OS2, ySuperscriptXSize, x);
-  case HB_OT_METRICS_SUPERSCRIPT_EM_Y_SIZE:       GET_METRICS (OS2, ySuperscriptYSize, y);
-  case HB_OT_METRICS_SUPERSCRIPT_EM_X_OFFSET:     GET_METRICS (OS2, ySuperscriptXOffset, x);
-  case HB_OT_METRICS_SUPERSCRIPT_EM_Y_OFFSET:     GET_METRICS (OS2, ySuperscriptYOffset, y);
-  case HB_OT_METRICS_STRIKEOUT_SIZE:              GET_METRICS (OS2, yStrikeoutSize, y);
-  case HB_OT_METRICS_STRIKEOUT_OFFSET:            GET_METRICS (OS2, yStrikeoutPosition, y);
-  case HB_OT_METRICS_UNDERLINE_SIZE:              GET_METRICS (post->table, underlineThickness, y);
-  case HB_OT_METRICS_UNDERLINE_OFFSET:            GET_METRICS (post->table, underlinePosition, y);
+  case HB_OT_METRICS_HORIZONTAL_ASCENDER:         return GET_METRICS (OS2, sTypoAscender, y) || GET_METRICS (hhea, ascender, y);
```

Comment:
I was thinking about this line becaome `return GET_METRICS (hmtx, asceder)`...

Reply:
Done